DumpFieldsMapCommand incompatible with symfony/dependency-injection ^6.4

### Bug Report

#### Summary

The bundle does not work when using `symfony/dependency-injection` `^6.4`.

This is because the `DumpFieldsMapCommand` (https://github.com/doctrine/DoctrineMongoDBBundle/blob/0f52b4db9ec106d19a41e1d3fe3e6e95e1416b19/src/Command/Encryption/DumpFieldsMapCommand.php#L38) expects an object implementing the `ServiceCollectionInterface` in the constructor.
But the 6.4 version of the Symfony dependency injection library does not implement this interface: https://github.com/symfony/dependency-injection/blob/5f311eaf0b321f8ec640f6bae12da43a14026898/ServiceLocator.php#L31
Compared to the 7.3 version, which does implement this interface: https://github.com/symfony/dependency-injection/blob/82119812ab0bf3425c1234d413efd1b19bb92ae4/ServiceLocator.php#L31

#### Current behavior

```
bin/console lint:container     

                                                                                                                        
 [ERROR] Invalid definition for service "doctrine_mongodb.odm.command.encryption_dump_fields_map": argument 1 of        
         "Doctrine\Bundle\MongoDBBundle\Command\Encryption\DumpFieldsMapCommand::__construct()" accepts                 
         "Symfony\Contracts\Service\ServiceCollectionInterface", "Symfony\Component\DependencyInjection\ServiceLocator" 
         passed.
```

#### Expected behavior

No error

#### How to reproduce

Install the following versions of Composer packages:
* `doctrine/mongodb-odm-bundle` `5.4.1`
* `symfony/dependency-injection` `v6.4.26`

--- a/config/command.php
+++ b/config/command.php
@@ -30,7 +30,7 @@ return static function (ContainerConfigurator $containerConfigurator): void {
 
         ->set('doctrine_mongodb.odm.command.encryption_dump_fields_map', DumpFieldsMapCommand::class)
             ->tag('console.command', ['command' => 'doctrine:mongodb:encryption:dump-fields-map'])
-            ->args([tagged_locator('doctrine_mongodb.odm.document_manager', 'name')])
+            ->args([service('doctrine_mongodb')])
 
         ->set('doctrine_mongodb.odm.command.create_schema', CreateSchemaDoctrineODMCommand::class)
             ->tag('console.command', ['command' => 'doctrine:mongodb:schema:create'])

--- a/src/Command/Encryption/DumpFieldsMapCommand.php
+++ b/src/Command/Encryption/DumpFieldsMapCommand.php
@@ -4,6 +4,7 @@ declare(strict_types=1);
 
 namespace Doctrine\Bundle\MongoDBBundle\Command\Encryption;
 
+use Doctrine\Bundle\MongoDBBundle\ManagerRegistry;
 use Doctrine\ODM\MongoDB\DocumentManager;
 use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
 use MongoDB\BSON\PackedArray;
@@ -14,8 +15,8 @@ use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
 use Symfony\Component\Yaml\Dumper;
-use Symfony\Contracts\Service\ServiceCollectionInterface;
 
+use function assert;
 use function json_decode;
 use function json_encode;
 use function sprintf;
@@ -34,8 +35,7 @@ use const JSON_UNESCAPED_UNICODE;
 )]
 final class DumpFieldsMapCommand extends Command
 {
-    /** @param ServiceCollectionInterface<DocumentManager> $documentManagers */
-    public function __construct(private readonly ServiceCollectionInterface $documentManagers)
+    public function __construct(private readonly ManagerRegistry $registry)
     {
         parent::__construct();
     }
@@ -60,7 +60,8 @@ final class DumpFieldsMapCommand extends Command
 
         $dumper = new Dumper();
 
-        foreach ($this->documentManagers as $name => $documentManager) {
+        foreach ($this->registry->getManagers() as $name => $documentManager) {
+            assert($documentManager instanceof DocumentManager);
             $encryptedFieldsMap = [];
             foreach ($documentManager->getMetadataFactory()->getAllMetadata() as $metadata) {
                 $database               = $documentManager->getDocumentDatabase($metadata->getName());

--- a/tests/DependencyInjection/DoctrineMongoDBExtensionTest.php
+++ b/tests/DependencyInjection/DoctrineMongoDBExtensionTest.php
@@ -8,6 +8,8 @@ use Closure;
 use Composer\InstalledVersions;
 use Composer\Semver\VersionParser;
 use Doctrine\Bundle\MongoDBBundle\Attribute\MapDocument;
+use Doctrine\Bundle\MongoDBBundle\Command\Encryption\DiagnosticCommand;
+use Doctrine\Bundle\MongoDBBundle\Command\Encryption\DumpFieldsMapCommand;
 use Doctrine\Bundle\MongoDBBundle\DependencyInjection\Compiler\ServiceRepositoryCompilerPass;
 use Doctrine\Bundle\MongoDBBundle\DependencyInjection\DoctrineMongoDBExtension;
 use Doctrine\Bundle\MongoDBBundle\Tests\DependencyInjection\Fixtures\Bundles\DocumentListenerBundle\EventListener\TestAttributeListener;
@@ -485,6 +487,34 @@ class DoctrineMongoDBExtensionTest extends TestCase
             ],
             $configuration->getMethodCalls(),
         );
+    }
+
+    public function testEncryptionCommands(): void
+    {
+        self::requireAutoEncryptionSupportInODM();
+
+        $container = $this->buildMinimalContainer();
+        $loader    = new DoctrineMongoDBExtension();
+
+        $config = [
+            'connections' => [
+                'default' => [
+                    'autoEncryption' => [
+                        'kmsProvider' => ['type' => 'local', 'key' => 'base64_encoded_key'],
+                    ],
+                ],
+            ],
+            'document_managers' => ['default' => []],
+        ];
+
+        $loader->load([$config], $container);
+        (new ServiceRepositoryCompilerPass())->process($container);
+
+        $dumpFieldsMapCommand = $container->get('doctrine_mongodb.odm.command.encryption_dump_fields_map');
+        $this->assertInstanceOf(DumpFieldsMapCommand::class, $dumpFieldsMapCommand);
+
+        $diagnosticCommand = $container->get('doctrine_mongodb.odm.command.encryption_diagnostic');
+        $this->assertInstanceOf(DiagnosticCommand::class, $diagnosticCommand);
     }
 
     public function testAutoEncryptionWithKeyVaultClientService(): void

